tests/test_examples.py

```diff
@@ -1276,6 +1282,53 @@ def test_second_stage_optimization():
     np.testing.assert_allclose(field[1].B0, 0, atol=1e-12)  # vertical field (vanishes)
 
 
+@pytest.mark.unit
+def test_second_stage_optimization_CoilSet():
```

Comment:
Are you able to check `jitable=True` and `jitable=False` somehow? Or maybe it's somewhere already and I don't see it

Reply:
It is implicitly inside of the compute function calls of the `QuadraticFlux` objective, sothe `jitable=True` part is checked, and we should not need to check the False case as that is never called inside of jit. The False cases are whenever we call `thing.compute` as `jitable=False` by default

Reply:
Are you sure this test doesn't already pass without these changes? Was the bug specific to `CoilSet`s and not individual `Coil`s? I'm curious why this bug is not a problem for `test_optimize_with_fourier_planar_coil` or any of the new tests Kian has added in PR #1025

Reply:
The error is because of `compute_magnetic_field`, where `curve.compute` is called. We had no tests with `QuadraticFlux` or any other objective which calls `coil.compute_magnetic_field` before now. If you don't believe it, try running this test on `master`, that should convince you